homeassistant/components/eufylife_ble/__init__.py

```diff
@@ -0,0 +1,70 @@
+"""The EufyLife integration."""
+from __future__ import annotations
+
+from eufylife_ble_client import EufyLifeBLEDevice
+
+from homeassistant.components import bluetooth
+from homeassistant.components.bluetooth.match import ADDRESS, BluetoothCallbackMatcher
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP, Platform
+from homeassistant.core import Event, HomeAssistant, callback
+
+from .const import CONF_MODEL, DOMAIN
+from .models import EufyLifeData
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up EufyLife device from a config entry."""
+    address = entry.unique_id
+    assert address is not None
+
+    model = entry.data[CONF_MODEL]
+    client = EufyLifeBLEDevice(model=model)
+
+    @callback
+    def _async_update_ble(
+        service_info: bluetooth.BluetoothServiceInfoBleak,
+        change: bluetooth.BluetoothChange,
+    ) -> None:
+        """Update from a ble callback."""
+        client.set_ble_device_and_advertisement_data(
+            service_info.device, service_info.advertisement
+        )
+        if not client.advertisement_data_contains_state:
+            hass.async_create_task(client.connect())
+
+    entry.async_on_unload(
+        bluetooth.async_register_callback(
+            hass,
+            _async_update_ble,
+            BluetoothCallbackMatcher({ADDRESS: address}),
+            bluetooth.BluetoothScanningMode.PASSIVE,
```

Comment:
Is the full name sent with passive scans?  Most devices don't do this and only send the short local name so its a bit surprising

Reply:
I'm not sure how to tell. I've tested this with ESPHome proxies and discovery is working perfectly. Do ESPHome proxies perform passive scanning?

Reply:
It depends if you have `active` set to `true` or `false` https://esphome.io/components/esp32_ble_tracker.html#configuration-variables

Reply:
I have it set to true. Should I try with it set to false?

Reply:
> I have it set to true. Should I try with it set to false?

Yes, you may find you get no names at all with `false`

Reply:
We try to get devices to work with out the passive data (its not always possible).  When its not the next best thing is to only require active scans for discovery and not for normal operation.

Some devices just don't send anything useful unless you have active scans though...

Reply:
You were right, once I set `active` to `false`, it stopped working. I updated the code to active scanning.